jetty-websocket/jetty-websocket-tests/src/test/java/org/eclipse/jetty/websocket/tests/server/FrameListenerTest.java

```diff
@@ -0,0 +1,202 @@
+    public void startServer() throws Exception
+    {
+        server = new Server();
+
+        ServerConnector connector = new ServerConnector(server);
+        server.addConnector(connector);
+
+        ServletContextHandler context = new ServletContextHandler();
+        context.setContextPath("/");
+
+        ServletHolder closeEndpoint = new ServletHolder(new WebSocketServlet()
+        {
+            @Override
+            public void configure(WebSocketServletFactory factory)
+            {
+                factory.getPolicy().setIdleTimeout(SECONDS.toMillis(2));
+                frameCreator = new FrameCreator();
+                factory.setCreator(frameCreator);
+            }
+        });
+        context.addServlet(closeEndpoint, "/ws");
+
+        HandlerList handlers = new HandlerList();
+        handlers.addHandler(context);
+        handlers.addHandler(new DefaultHandler());
+
+        server.setHandler(handlers);
+
+        server.start();
+    }
+
+    @AfterEach
+    public void stopServer() throws Exception
+    {
+        server.stop();
+    }
+
+    @BeforeEach
+    public void startClient() throws Exception
+    {
+        client = new WebSocketClient();
+        client.start();
+    }
+
+    @AfterEach
+    public void stopClient() throws Exception
+    {
+        client.stop();
+    }
+
+    private void close(Session session)
+    {
+        if (session != null)
+        {
+            session.close();
+        }
+    }
+
+    @Test
+    public void testPartialText() throws Exception
+    {
+        ClientUpgradeRequest request = new ClientUpgradeRequest();
+        CloseTrackingEndpoint clientEndpoint = new CloseTrackingEndpoint();
+
+        URI wsUri = WSURI.toWebsocket(server.getURI().resolve("/ws"));
+        Future<Session> futSession = client.connect(clientEndpoint, wsUri, request);
+
+        Session session = null;
+        try (StacklessLogging ignore = new StacklessLogging(WebSocketSession.class))
+        {
+            session = futSession.get(5, SECONDS);
+
+            RemoteEndpoint clientRemote = session.getRemote();
+            clientRemote.sendPartialString("hello", false);
+            clientRemote.sendPartialString(" ", false);
+            clientRemote.sendPartialString("world", true);
+
+            FrameEndpoint serverEndpoint = frameCreator.frameEndpoint;
+
+            String event = serverEndpoint.frameEvents.poll(5, SECONDS);
+            assertThat("Event", event, is("FRAME[TEXT,fin=false,payload=hello,len=5]"));
+            event = serverEndpoint.frameEvents.poll(5, SECONDS);
+            assertThat("Event", event, is("FRAME[CONTINUATION,fin=false,payload= ,len=1]"));
+            event = serverEndpoint.frameEvents.poll(5, SECONDS);
+            assertThat("Event", event, is("FRAME[CONTINUATION,fin=true,payload=world,len=5]"));
+        }
+        finally
+        {
+            close(session);
+        }
+    }
+
+    public static class FrameCreator implements WebSocketCreator
```

Comment:
might be cleaner to use a lambda for the creator
```java
Private FrameEndpoint serverEndpoint = new FrameEndpoint();
...
factory.setCreator((req, res) -> serverEndpoint);
```

Reply:
Fixed in commit b8f29630eddd9575ad142dc9821c54a2d26f16c0